### PR TITLE
scx_layered: Refactor layered initialization

### PR DESCRIPTION
**Pull request by hodgesds**
Refactor layer initialization into a separate helper function.

### CODE REVIEW

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 1]
```diff
@@ -2559,126 +2680,11 @@ s32 BPF_STRUCT_OPS_SLEEPABLE(layered_init)
...
-		 */
-		bpf_cpumask_setall(cpumask);
-
-		cpumask = bpf_kptr_xchg(&cpumaskw->cpumask, cpumask);
-		if (cpumask)
-			bpf_cpumask_release(cpumask);
-
-		// create the dsqs for the layer
-		bpf_for(j, 0, nr_llcs) {
-			int node_id = llc_node_id(i);
```
Comment (hodgesds) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:2673]: This looks like a really bad bug.... the llc node id was being looked up based on the layer id and not the LLC.

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 2]
```diff
@@ -2558,127 +2678,13 @@ s32 BPF_STRUCT_OPS_SLEEPABLE(layered_init)
 	dbg("CFG: Dumping configuration, nr_online_cpus=%d smt_enabled=%d little_cores=%d",
 	    nr_online_cpus, smt_enabled, has_little_cores);
 
+	u64 fallback_dsq_id = 0;
```
Comment (likewhatevs) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:2681]: nit: shouldn't that be some FALLBACK_DSQ_ID defined somewhere?

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 3]
```diff
@@ -2452,12 +2452,132 @@ struct layered_timer layered_timers[MAX_TIMERS] = {
 // TODO: separate this out to a separate compilation unit
 #include "timer.bpf.c"
 
+static s32 init_layer(int layer_id, u64 *fallback_dsq_id)
```
Comment (likewhatevs) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:2455]: nit: a nice well formatted comment here would be nice

**Review by likewhatevs — APPROVED** [review]
LGTM I think

File: scheds/rust/scx_layered/src/bpf/main.bpf.c (hunk 2 above)
Comment (hodgesds) [reply scheds/rust/scx_layered/src/bpf/main.bpf.c:2684]: So this is just the counter that is used to sequentially allocate the hi fallback DSQs.

**Pull request merged**